clippy_lints/src/loops/while_immutable_condition.rs

```diff
@@ -64,20 +65,18 @@ struct HasBreakOrReturnVisitor {
 }
 
 impl<'tcx> Visitor<'tcx> for HasBreakOrReturnVisitor {
-    fn visit_expr(&mut self, expr: &'tcx Expr<'_>) {
-        if self.has_break_or_return {
-            return;
-        }
-
+    type Result = ControlFlow<()>;
+    fn visit_expr(&mut self, expr: &'tcx Expr<'_>) -> ControlFlow<()> {
         match expr.kind {
             ExprKind::Ret(_) | ExprKind::Break(_, _) => {
                 self.has_break_or_return = true;
```

Comment:
I think we can even remove this bool from the struct, by using `.is_break()` on the visit result

Reply:
same in option_map_unwrap_or.rs and lifetimes.rs